### ISSUE TITLE
will Calcpad support file reading and wrting in the future

### ISSUE THREAD

**Issue by idealkindom** [Proektsoftbg/Calcpad#100]
Reading ang writing data file is very useful, hoping Calcpad could do this in the future.

**Comment by Proektsoftbg** [comment]
Yes, but after we add vectors and matrices. :) We need to have where to store the data.
What we can do quickly is to add an "Import from Excel" command. Calcpad will read the cells and insert the contents into rows, warped in take functions:
```
R1(j) = take(j; A1; B1; C1...)
R2(j) = take(j; A2; B2; C2...)
R3(j) = take(j; A3; B3; C3...)
M(i;j) = take(j; R1(j) ; R2(j) ; R3(j) ...)
```
Then you will be able to use the data as matrix. We can even make this interactive - to open the spreadsheet and ask the user to select a cell range.

**Comment by idealkindom** [comment]
> Yes, but after we add vectors and matrices. :) We need to have where to store the data. What we can do quickly is to add an "Import from Excel" command. Calcpad will read the cells and insert the contents into rows, warped in take functions:
> 
> ```
> R1(j) = take(j; A1; B1; C1...)
> R2(j) = take(j; A2; B2; C2...)
> R3(j) = take(j; A3; B3; C3...)
> M(i;j) = take(j; R1(j) ; R2(j) ; R3(j) ...)
> ```
> 
> Then you will be able to use the data as matrix. We can even make this interactive - to open the spreadsheet and ask the user to select a cell range.

Hi, Ned. I don't have knowledge and skills on systematic programming, instead I used Maxima and GNU Octave for many years on engineering calculation.

I'm wondering is it possilbe to learn to contribute to the development of Calcpad.
Would you please give me some suggestions how to pick up? it would be very kind of you if some tutorials for development could be advised.
I think another important need for Calcpad  is to improve the function of plot, i.e. mutiplot with label, legend , line style and symbol.Also,Though map is very good, 3d plot is still attractive.

**Comment by Proektsoftbg** [comment]
Hi, Thank you!
I will consider this.

About plotting, it is

**Comment by hildebrandopsj** [comment]
> Yes, but after we add vectors and matrices. :) We need to have where to store the data. What we can do quickly is to add an "Import from Excel" command. Calcpad will read the cells and insert the contents into rows, warped in take functions:
> 
> ```
> R1(j) = take(j; A1; B1; C1...)
> R2(j) = take(j; A2; B2; C2...)
> R3(j) = take(j; A3; B3; C3...)
> M(i;j) = take(j; R1(j) ; R2(j) ; R3(j) ...)
> ```
> 
> Then you will be able to use the data as matrix. We can even make this interactive - to open the spreadsheet and ask the user to select a cell range.

Interact with excel will be amazing! I can see Calcpad working side by side with SAP2000 ;)

**Comment by Proektsoftbg** [comment]
I tried to create a macro function for 3D plotting with Plotly. The result is not so bad, but works slow and throws some script errors sometimes. I will continue to experiment further and when ready, I will post the solution. Maybe, we can create a macro library that can be used like follows:
```
#include Plot.cpd
Plot3D$(name$; function$; width$; height$)
```
![image](https://user-images.githubusercontent.com/90918363/203925301-b4533b94-2440-451e-9fc0-61efb58c8fef.png)
The new "macros" and "modules" features in Calcpad provide unlimited possibilites to extend its functionality. :)

**Comment by Proektsoftbg** [comment]
Yes, currently it requires internet to work, but after we select the appropriate platform and implement the macro interface, we can download the library as local files and add it to the Calcpad setup.
